### PR TITLE
Configurator Unit Tests

### PR DESCRIPTION
**Pull request by Chase-Grajeda**
Added basic unit tests to the `Configurator` experimental class, focusing on initialization and correct shape output from the respective configuration filter methods.

The following tests are currently implemented:
- Configurator initialization with all arguments
- Configurator initialization with only required arguments
- Inference variable filter returns correct shape
- Inference condition filter returns correct shape without `summary_outputs`
- Inference condition filter returns correct shape when `summary_outputs` is present in `data`
- Inference condition filter returns correct shape when `summary_outputs` is present in `data` and `keys`
- Summary variable filter returns correct shape
- Summary conditions filter returns correct shape
- Inference condition, summary variable, and summary condition filters return None for missing `keys`

Feel free to recommend more tests for the suite!

### CODE REVIEW

File: tests/test_configurators/test_configurators.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    assert config.inference_conditions == []
+    assert config.summary_variables == []
+    assert config.summary_conditions == []
+
+
+# Test successful configure_inference_variables()
+def test_inference_vars_filter(random_data, configurator: Configurator, test_shape):
+    config = configurator
+    filtered_data = config.configure_inference_variables(random_data)
+    assert filtered_data.shape == test_shape
```
Comment (LarsKue) [inline tests/test_configurators/test_configurators.py:100]: This test only works if there is only a single inference_variable, i.e. no concatenation actually happens

File: tests/test_configurators/test_configurators.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+
+# Test return None for filters when configuring sparse Configurator
+def test_null_vars_and_conds(random_data_no_output, configurator_sparse: Configurator):
+    config = configurator_sparse
+    filtered_inference_conds = config.configure_inference_conditions(random_data_no_output)
+    filtered_summary_vars = config.configure_summary_variables(random_data_no_output)
+    filtered_summary_conds = config.configure_summary_conditions(random_data_no_output)
+    assert filtered_inference_conds == None
```
Comment (LarsKue) [inline tests/test_configurators/test_configurators.py:144]: use `is None`

File: tests/test_configurators/test_configurators.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+@pytest.fixture()
+def configurator_sparse(test_params):
+    return Configurator(
+        inference_variables=test_params['inference_variables'],
+    )
+
+
+# Test for correct construction of Configurator with all args
+def test_configurator_init(test_params, configurator: Configurator):
+    config = configurator
```
Comment (LarsKue) [inline tests/test_configurators/test_configurators.py:80]: This renaming has some repeat code smell.

**Review by LarsKue — CHANGES_REQUESTED** [review]
Thank you for the PR! I think this is a solid basis, but we need to be a little bit more thorough with the test parametrization.

File: tests/test_configurators/test_configurators.py
[the remark this answers is not in the document]
Comment (Chase-Grajeda) [reply tests/test_configurators/test_configurators.py:1]: Resolved in a154911

File: tests/test_configurators/test_configurators.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
+import keras
+import pytest
+
+from bayesflow.experimental.configurators import Configurator
+
+
+@pytest.fixture()
+def test_shape():
+    return (5,7)
```
[the remark this answers is not in the document]
Comment (Chase-Grajeda) [reply tests/test_configurators/test_configurators.py]: Added to `conftest.py` and parameterized over `[2,3]` (see b2922ca)

File: tests/test_configurators/test_configurators.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+def random_data_no_output(test_shape):
+    return {
+        'var1': keras.random.normal(test_shape),
+        'var2': keras.random.normal(test_shape),
+        'var3': keras.random.normal(test_shape),
+    }
+
+
+@pytest.fixture()
+def test_params():
```
[the remark this answers is not in the document]
Comment (Chase-Grajeda) [reply tests/test_configurators/test_configurators.py]: Parameterized over `[True, False]`  (see b2922ca)

File: tests/test_configurators/test_configurators.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+@pytest.fixture()
+def test_shape():
+    return (5,7)
+
+
+@pytest.fixture()
+def random_data(test_shape):
+    return {
+        'var1': keras.random.normal(test_shape),
```
[the remark this answers is not in the document]
Comment (Chase-Grajeda) [reply tests/test_configurators/test_configurators.py]: Strings using single quotes now use double quotes (see b2922ca)

File: tests/test_configurators/test_configurators.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+@pytest.fixture()
+def configurator_sparse(test_params):
+    return Configurator(
+        inference_variables=test_params['inference_variables'],
+    )
+
+
+# Test for correct construction of Configurator with all args
+def test_configurator_init(test_params, configurator: Configurator):
```
[the remark this answers is not in the document]
Comment (Chase-Grajeda) [reply tests/test_configurators/test_configurators.py]: Fixture type hints removed in `test_configurators.py` (see b2922ca)

File: tests/test_configurators/test_configurators.py [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+    assert config.summary_conditions == test_params['summary_conditions']
+
+
+# Test for correct construction of Configurator with only inference_vars
+def test_sparse_configurator_init(test_params, configurator_sparse: Configurator):
+    config = configurator_sparse
+    assert config.inference_variables == test_params['inference_variables']
+    assert config.inference_conditions == []
+    assert config.summary_variables == []
+    assert config.summary_conditions == []
```
[the remark this answers is not in the document]
Comment (Chase-Grajeda) [reply tests/test_configurators/test_configurators.py]: Test removed (see b2922ca)

File: tests/test_configurators/test_configurators.py [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+
+
+@pytest.fixture()
+def random_data(test_shape):
+    return {
+        'var1': keras.random.normal(test_shape),
+        'var2': keras.random.normal(test_shape),
+        'var3': keras.random.normal(test_shape),
+        'summary_outputs': keras.random.normal(test_shape),
+    }
```
[the remark this answers is not in the document]
Comment (Chase-Grajeda) [reply tests/test_configurators/test_configurators.py]: Added `"summary_inputs"` and `"summary_conditions"` to the `random_data` fixture in `conftest.py` (see b2922ca). Is this what you were looking for?

File: tests/test_configurators/test_configurators.py (hunk 1 above)
Comment (Chase-Grajeda) [reply tests/test_configurators/test_configurators.py]: Refactored test to account for multiple tensors (see b2922ca)

File: tests/test_configurators/test_configurators.py (hunk 3 above)
Comment (Chase-Grajeda) [reply tests/test_configurators/test_configurators.py]: Removed `config = configurator` instances and used the respective function arg instead (see b2922ca)

**Review by LarsKue — APPROVED** [review]
LGTM! Thank you for the PR

**Pull request merged**